modules/oxide-components/src/test/ts/browser/keynav/KeynavFlowTypeTest.spec.tsx

```diff
@@ -0,0 +1,245 @@
+    background-color: red;
+  }
+`;
+
+const store: string[] = [];
+
+const clearStore = (store: string[]) => {
+  store.splice(0, store.length);
+};
+
+interface ItemProps {
+  classes: string[];
+  name: string;
+  store: string[];
+}
+
+const Item = ({ classes, name, store }: ItemProps) =>
+  (
+    <span
+      className={classes.join(' ')}
+      style={{
+        display: 'inline-block',
+        width: '20px',
+        height: '20px',
+        margin: '2px',
+        border: '1px solid ' + (classes.includes('stay') ? 'blue' : 'yellow')
+      }}
+      tabIndex={-1}
+      onClick={() => store.push('item.execute: ' + name)}
+    />
+  );
+
+const Container = forwardRef<HTMLDivElement, { store: string[]; testId: string }>(({ store, testId }, ref) => (
+  <div
+    data-testid={testId}
+    ref={ref}
+    className="flow-keying-test"
+    style={{
+      background: 'white',
+      width: '200px',
+      height: '200px'
+    }}
+  >
+    <Item classes={[ 'stay', 'one' ]} name="one" store={store} />
+    <Item classes={[ 'stay', 'two' ]} name="two" store={store} />
+    <Item classes={[ 'skip', 'three' ]} name="three" store={store} />
+    <Item classes={[ 'skip', 'four' ]} name="four" store={store} />
+    <Item classes={[ 'stay', 'five' ]} name="five" store={store} />
+  </div>
+));
+
+const sequence = async (doc: Document, key: string, identifiers: Array<{ label: string; selector: string }>) => {
+  for (let i = 0; i < identifiers.length; i++) {
+    await userEvent.keyboard(`{${key}}`);
+    const element = doc.querySelector(identifiers[i].selector);
+    await expect.element(element, {
+      message: 'Focus should move from ' + (i > 0 ? identifiers[i - 1].label : '(start)') + ' to ' + identifiers[i].label
+    }).toHaveFocus();
+  }
+};
+
+const targets = {
+  one: { label: 'one', selector: '.one' },
+  two: { label: 'two', selector: '.two' },
+  five: { label: 'five', selector: '.five' }
+};
+
+describe('KeynavFowTypeTest', () => {
+
+  beforeEach(() => {
+    clearStore(store);
+  });
+
+  test('FlowKeying with cycles', async () => {
+
+    const FlowKeyingWithCycles = () => {
+      const ref = useRef<HTMLDivElement>(null);
+      useFlowKeyNavigation({
+        containerRef: ref,
+        selector: '.stay',
+        escape: () => {
+          store.push('flow.onEscape');
+          return Optional.some(true);
+        },
+        cycles: true,
+      });
+      return (
+        <Container ref={ref} store={store} testId='container' />
+      );
+    };
+
+    const { getByTestId, baseElement } = render(<FlowKeyingWithCycles />, {
+      wrapper: ({ children }) => (
+        <>
+          <style>
+            {styles}
+          </style>
+          <div className='tox'>
+            {children}
+          </div>
+        </>
+      )
+    });
+    const doc = baseElement.ownerDocument;
+    const container = getByTestId('container');
+    const firstChild: HTMLElement | null = container.element().querySelector('.one');
```

Comment:
Don't we want o blow up if this fails?

Reply:
It will blow up 2 lines down the line, but this is more like the `getOrDie` we use in tests that should never fail

Reply:
Is the `one` expected to always exist. Then this isn't safe enough. Then we should have a querySelectorOrDie abstraction or something on top of this. If I alter `one` to `oneX` then it fails on focusing not on finding the element is that expected?

Reply:
We can extend them quite easily: https://vitest.dev/guide/browser/locators.html#custom-locators